debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorMetadata.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.apache.kafka.connect.source.SourceConnector;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Field;
+import io.debezium.heartbeat.Heartbeat;
+import io.debezium.metadata.AbstractConnectorMetadata;
+import io.debezium.metadata.ConnectorDescriptor;
+import io.debezium.metadata.ConnectorProperty;
+import io.debezium.metadata.PropertyMetadata;
+import io.debezium.schema.DataCollectionId;
+
+// @TODO move metadata into Field?
```

Comment:
Should we move the metadata into the Fields in the ConnectorConfig?

Alternatively restructure validators, recommenders, etc to abstract concepts like isRequired/isMandator, #allowedValues instead of EnumRecommender, etc. ?

Reply:
@gunnarmorling the basic decision was to extend the Field class with the metadata for the UI API.

But I have some detail questions:

- `io.debezium.config.Field#withEnum()` currently only adds an `EnumRecommender` as `Validator` to test for allowed values, I would add storing the allowed values from the enum as well, are you fine with it?

- The information if the field is mandatory/required is currently coming from a Validator, is it okay to extend the `Field` DSL with an `#isMandatory()` method that adds the validator and stores that state so I can retrieve it for the API?

- Currently we have 4 groups for the Kafka `ConfigDef` in `io.debezium.config.ConfigDefinition#configDef()` which are: one group for the specific `connectorName` (provided in the ConnectorConfig with help of the `ConfigDefinitionEditor`, eg `"MySQL"`), `"Connector`", `"History Storage"` and `"Events"`.  We could extend this and use the API categories here, or add an additional property to the Field implementation. That would imply changing `ConnectorDefinition` and `ConnectorDefinitionEditor` as well and maybe all `*ConnectorConfig` implementations might be affected. WDYT?

Reply:
I created https://issues.redhat.com/browse/DBZ-3446 and https://issues.redhat.com/browse/DBZ-3447 as follow-up tasks.